packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,445 @@
+    }
+
+    handleTabIndex();
+  }, [children, isCollapsible, handleTabIndex, generateRollUp]);
+
+  useLayoutEffect(() => {
+    let observer = new ResizeObserver((entries) => {
+      handleResize(entries);
+    });
+    observer.observe(filterPillsRef.current.parentNode);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, [children, handleResize, isCollapsed]);
+
+  const focusPillsContainer = () => {
+    setContainerTabindex('0');
+    filterPillsRef.current.focus();
+  };
+
+  // When a pill is deleted, focuses the new pill.
+  useEffect(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const startIndex = PillsUtils.handleRollUp(filterPillsRef);
+
+    if (isPillDeleted.current) {
+      if (React.Children.count(children) <= 0) {
+        focusPillsContainer();
+        return;
+      }
+      // if there is a roll Up pill, set tabindex to 0
+      if (rollUpPill) {
+        PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+      }
+
+      if (pills.length > 0) {
+        PillsUtils.setPillsTabIndex(pills, '-1');
+      }
+
+      if (pills.length === startIndex) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill && startIndex >= 1) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill) {
+        currentPill.current = rollUpPill.getAttribute('id');
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+      isPillDeleted.current = false;
+    }
+  }, [isPillDeleted, children]);
+
+  useEffect(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    // To focus the immediate focusable pill after the rollUp pill is selected/removed
+    if (isCollapsible && !isCollapsed) {
+      if (pills.length === React.Children.count(children)) {
+        setTabIndex('-1');
+        focusNode.current += 1;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+        if (isRollUpRemoved.current) {
+          focusCurrentNode();
+          isRollUpRemoved.current = false;
+        }
+      }
+    }
+  }, [children, isCollapsed, isCollapsible, updatedCount]);
+
+  const focusNextNode = (pills, rollUpPill) => {
+    // if the next pill is roll up pill, focus the roll up pill
+
+    if (focusNode.current + 1 === pills.length || focusNode.current + 1 < pills.length) {
+      setTabIndex('-1');
+      if (rollUpPill && focusNode.current + 1 === pills.length) {
+        currentPill.current = rollUpPill.getAttribute('id');
+      } else if (focusNode.current + 1 < pills.length) {
+        focusNode.current += 1;
+        currentPill.current = children[focusNode.current].props.id;
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const focusPreviousNode = (pills, rollUpPill) => { // lp052179
```

Comment:
Remove `lp052179`.

Reply:
Removed in this commit e79be980bb3defef6e9370f26558523eead90bfa